python/demo/gsoc2022/demo_scattering_boundary_conditions.py

```diff
@@ -0,0 +1,451 @@
+# \bar{\mathbf{v}}+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s \cdot
+# \bar{\mathbf{v}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{v}}~\mathrm{d}x \\ +& \int_{\partial \Omega}
+# (\mathbf{n} \times \nabla \times \mathbf{E}_s) \cdot \bar{\mathbf{v}}
+# +\left(j n_bk_{0}+\frac{1}{2r}\right) (\mathbf{n} \times \mathbf{E}_s
+# \times \mathbf{n}) \cdot \bar{\mathbf{v}}~\mathrm{d}s=0
+# \end{align}.
+# $$
+#
+# By using the $(\nabla \times \mathbf{A}) \cdot \mathbf{B}=\mathbf{A}
+# \cdot(\nabla \times \mathbf{B})+\nabla \cdot(\mathbf{A}
+# \times \mathbf{B})$
+# relation, we can change the first term into:
+#
+# $$
+# \begin{align}
+# & \int_{\Omega}-\nabla \cdot(\nabla\times\mathbf{E}_s \times
+# \bar{\mathbf{v}})-\nabla \times \mathbf{E}_s \cdot \nabla
+# \times\bar{\mathbf{v}}+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s
+# \cdot \bar{\mathbf{v}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{v}}~\mathrm{dx} \\ +&\int_{\partial \Omega}
+# (\mathbf{n} \times \nabla \times \mathbf{E}_s) \cdot \bar{\mathbf{v}}
+# +\left(j n_bk_{0}+\frac{1}{2r}\right) (\mathbf{n} \times \mathbf{E}_s
+# \times \mathbf{n}) \cdot \bar{\mathbf{v}}~\mathrm{d}s=0,
+# \end{align}
+# $$
+#
+# using the divergence theorem $\int_\Omega\nabla\cdot\mathbf{F}~\mathrm{d}x =
+# \int_{\partial\Omega} \mathbf{F}\cdot\mathbf{n}~\mathrm{d}s$, we can write:
+#
+# $$
+# \begin{align}
+# & \int_{\Omega}-(\nabla \times \mathbf{E}_s) \cdot (\nabla \times
+# \bar{\mathbf{v}})+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s \cdot
+# \bar{\mathbf{v}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{v}}~\mathrm{d}x \\ +&\int_{\partial \Omega}
+# -(\nabla\times\mathbf{E}_s \times \bar{\mathbf{v}})\cdot\mathbf{n}
+# + (\mathbf{n} \times \nabla \times \mathbf{E}_s) \cdot \bar{\mathbf{v}}
+# +\left(j n_bk_{0}+\frac{1}{2r}\right) (\mathbf{n} \times \mathbf{E}_s
+# \times \mathbf{n}) \cdot \bar{\mathbf{v}}~\mathrm{d}s=0.
+# \end{align}
+# $$
+#
+# We can cancel $-(\nabla\times\mathbf{E}_s \times \bar{\mathbf{V}})
+# \cdot\mathbf{n}$  and $\mathbf{n} \times \nabla \times \mathbf{E}_s
+# \cdot \bar{\mathbf{V}}$ thanks to the triple product rule $\mathbf{A}
+# \cdot(\mathbf{B} \times \mathbf{C})=\mathbf{B} \cdot(\mathbf{C} \times
+# \mathbf{A})=\mathbf{C} \cdot(\mathbf{A} \times \mathbf{B})$, arriving
+# at the final weak form:
+#
+# $$
+# \begin{align}
+# & \int_{\Omega}-(\nabla \times \mathbf{E}_s) \cdot (\nabla \times
+# \bar{\mathbf{v}})+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s \cdot
+# \bar{\mathbf{v}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{v}}~\mathrm{d}x \\ +&\int_{\partial \Omega}
+# \left(j n_bk_{0}+\frac{1}{2r}\right)( \mathbf{n} \times \mathbf{E}_s \times
+# \mathbf{n}) \cdot \bar{\mathbf{v}} ~\mathrm{d} s = 0.
+# \end{align}
+# $$
+#
+# ## Implementation
+# The modules that will be used are imported:
+
+# +
+import numpy as np
```

Comment:
```suggestion
import gmsh
import numpy as np
```
Required for finalization of gmsh after generating mesh